File: app/src/main/java/com/app/rupiksha/adapters/BannerAdapter.java
package com.app.rupiksha.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;


import com.bumptech.glide.Glide;
import com.app.rupiksha.R;
import com.app.rupiksha.databinding.BannerRecycleviewListBinding;
import com.app.rupiksha.models.BannerModel;

import java.util.List;

public class BannerAdapter extends RecyclerView.Adapter<BannerAdapter.ViewHolder> {
    private Context context;
    private List<BannerModel> list;
    BannerAdapter.OnItemClick onItemClick;

    public BannerAdapter(Context context, List<BannerModel> listAnnonce, OnItemClick onItemClick) {
        this.context = context;
        this.list = listAnnonce;
        this.onItemClick=onItemClick;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        BannerRecycleviewListBinding annoncesRecycleviewListBinding = DataBindingUtil.bind(LayoutInflater.from(parent.getContext()).inflate(R.layout.banner_recycleview_list, parent, false));
        return new ViewHolder(annoncesRecycleviewListBinding);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position)
    {

        if(!list.get(position).getFilepath().isEmpty())
        {
            Glide.with(context).load(list.get(position).getFilepath()).placeholder(R.color.white_60).into(holder.binding.image);
        }
        holder.binding.startView.setVisibility(position == 0 ? View.VISIBLE : View.GONE);
        holder.binding.endView.setVisibility(position == list.size() - 1 ? View.VISIBLE : View.GONE);

    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        private BannerRecycleviewListBinding binding;

        public ViewHolder(@NonNull BannerRecycleviewListBinding itemView) {
            super(itemView.getRoot());
            this.binding = itemView;
        }
    }
    public static interface OnItemClick {
        public void onClick(int position,int proUserId, String link);
    }
}

File: app/src/main/java/com/app/rupiksha/adapters/BannerAdapter.kt
package com.app.rupiksha.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.app.rupiksha.R
import com.app.rupiksha.databinding.BannerRecycleviewListBinding
import com.app.rupiksha.models.BannerModel
import com.bumptech.glide.Glide
import com.bumptech.glide.request.RequestOptions

class BannerAdapter(
    private val context: Context,
    private val list: List<BannerModel>,
    private val onItemClick: OnItemClick? = null
) : RecyclerView.Adapter<BannerAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val binding: BannerRecycleviewListBinding = DataBindingUtil.inflate(
            LayoutInflater.from(parent.context),
            R.layout.banner_recycleview_list,
            parent,
            false
        )
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val model = list[position]

        if (!model.filepath.isNullOrEmpty()) {
            Glide.with(context)
                .load(model.filepath)
                .placeholder(R.color.white_60)
                .into(holder.binding.image)
        }
        
        holder.binding.startView.visibility = if (position == 0) View.VISIBLE else View.GONE
        holder.binding.endView.visibility = if (position == list.size - 1) View.VISIBLE else View.GONE
    }

    override fun getItemCount(): Int = list.size

    inner class ViewHolder(val binding: BannerRecycleviewListBinding) : RecyclerView.ViewHolder(binding.root)

    interface OnItemClick {
        fun onClick(position: Int, proUserId: Int, link: String)
    }
}
